src/gamepad/linux/linux-raw-gamepad-monitor.vala

```diff
@@ -0,0 +1,96 @@
+// This file is part of GNOME Games. License: GPLv3
+
+private class LibGamepad.LinuxRawGamepadMonitor : Object, RawGamepadMonitor {
+	public delegate void RawGamepadCallback (RawGamepad raw_gamepad);
+
+	private static LinuxRawGamepadMonitor instance;
+
+	private GUdev.Client client;
+	private HashTable<string, RawGamepad> raw_gamepads;
+
+	private LinuxRawGamepadMonitor () {
+		client = new GUdev.Client ({"input"});
+		client.uevent.connect (handle_udev_client_callback);
+
+		// Initialize initially plugged in gamepads
+		var initial_devices_list = client.query_by_subsystem ("input");
+		foreach (var device in initial_devices_list) {
+			if (device.get_device_file () == null)
+				continue;
+
+			if (!is_gamepad (device))
+				continue;
+
+			add_gamepad (device);
+		}
+	}
+
+	public static LinuxRawGamepadMonitor get_instance () {
+		if (instance == null)
+			instance = new LinuxRawGamepadMonitor ();
+
+		return instance;
+	}
+
+	public void foreach_gamepad (RawGamepadCallback callback) {
+		raw_gamepads.foreach((identifier, raw_gamepad) => callback (raw_gamepad));
+	}
+
+	private void handle_udev_client_callback (string action, GUdev.Device device) {
+		if (device.get_device_file () == null)
+			return;
+
+		if (!is_gamepad (device))
+			return;
+
+		switch (action) {
+		case "add":
+			var raw_gamepad = add_gamepad (device);
+			if (raw_gamepad != null)
+				gamepad_plugged (raw_gamepad);
+
+			break;
+		case "remove":
+			var raw_gamepad = remove_gamepad (device);
+			if (raw_gamepad != null) {
```

Comment:
Even though there are multiple lines, there is only one statement so we can remove the braces. Sorry if I wasn't clear about this, it is explained here: https://github.com/Kekun/gnome-games/blob/master/CodingStyle.txt#L48